src/main/resources/templates/list.html

```diff
@@ -136,6 +136,14 @@ <h4>Event-Infos</h4>
             selectedRegistrations.splice(selectedRegistrations.findIndex(e => e === elem.value), 1);
         }
     }
+    $('#select-all').click(function(event) {
+        const selectAll = this;
+        $(':checkbox').each(function() {
+            if (this === selectAll) return;
```

Comment:
Prinzipiell funktioniert das mit dem `this`, ich bin da aber kein großer Fan von. Deine Checkbox hat ja schon eine ID (`select-al`), darauf könntest Du überprüfen.